backend/compact-connect/lambdas/common-python/cc_common/data_model/schema/provider.py

```diff
@@ -95,3 +105,22 @@ def populate_fam_giv_mid(self, in_data, **kwargs):  # noqa: ARG001 unused-argume
     def drop_fam_giv_mid(self, in_data, **kwargs):  # noqa: ARG001 unused-argument
         del in_data['providerFamGivMid']
         return in_data
+
+    @pre_load
+    def pre_load_initialization(self, in_data, **kwargs):  # noqa: ARG001 unused-argument
+        return self._calculate_status(in_data)
+
+    def _calculate_status(self, in_data, **kwargs):
+        # determine if the status is active or inactive by checking the jurisdictionStatus
+        # along with the expiration date of the license
+        in_data['status'] = (
+            'active'
+            if (
+                in_data['jurisdictionStatus'] == 'active'
+                and date.fromisoformat(in_data['dateOfExpiration'])
+                > datetime.now(tz=config.expiration_date_resolution_timezone).date()
+            )
+            else 'inactive'
+        )
```

Comment:
Any easy way to reduce duplication of `remove_status_field_if_present`, `pre_load_initialization`, and `_calculate_status` across the license, privilege, and provider models?

Reply:
@jlkravitz Great call. I've introduced a common schema which handles this common serialization/deserialization logic and updated the license, privilege, and provider models to inherit it.